frontend/packages/ceph-storage-plugin/src/components/independent-mode/install.tsx

```diff
@@ -76,42 +47,33 @@ const InstallExternalCluster = withHandlePromise((props: InstallExternalClusterP
     const file = event.target.files[0];
     reader.onload = (ev) => {
       const data = ev.target?.result as string;
+      if (isValidJSON(data)) {
+        setDataError(checkError(fileData, minKeys));
+      } else {
+        setDataError('The uploaded file is not a valid JSON file');
+      }
       setFileData(data);
     };
     reader.readAsText(file);
   };
 
   const onSubmit = (event) => {
     event.preventDefault();
-    // https://github.com/rook/rook/blob/master/cluster/examples/kubernetes/ceph/import-external-cluster.sh
+
     const secret = {
       apiVersion: SecretModel.apiVersion,
       kind: SecretModel.kind,
       metadata: {
-        name: 'rook-ceph-mon',
+        name: 'rook-ceph-external-cluster-details',
         namespace: ns,
       },
       stringData: {
-        'cluster-name': clusterName,
-        fsid: externalFSID,
-        'admin-secret': externalAdminSecret,
-        'mon-secret': 'mon-secret',
+        // eslint-disable-next-line @typescript-eslint/camelcase
+        external_cluster_details: fileData,
```

Comment:
filedata needs to be base64 encoded.
```suggestion
        external_cluster_details: fileData,
```

Reply:
Kubernetes performs Base64 encoding.